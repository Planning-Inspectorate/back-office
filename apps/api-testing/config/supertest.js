// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
