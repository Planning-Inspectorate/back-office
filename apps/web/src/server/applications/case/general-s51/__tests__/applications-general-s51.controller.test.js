// @ts-nocheck
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../testing/index.js';
import {
	generalSection51CaseReference,
	generalSection51FolderName
} from '../applications-general-s51.config.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const nocks = () => {
	nock('http://test/')
		.persist()
		.get(`/applications/reference/${generalSection51CaseReference}`)
		.reply(200, { id: 111111111, reference: generalSection51CaseReference });

	nock('http://test/')
		.persist()
		.get(`/applications/111111111/folders`)
		.reply(200, [{ id: 222222222, displayNameEn: generalSection51FolderName }]);

	nock('http://test/')
		.persist()
		.get(`/applications-service/case/111111111/project-documentation/222222222/s51-advice`)
		.reply(200, {});
};

describe('General section 51 page', () => {
	beforeEach(installMockApi);

	afterEach(() => {
		nock.cleanAll();
		teardown();
	});

	describe('#viewGeneralSection51page', () => {
		beforeEach(async () => {
			nocks();
		});

		it('should redirect directly to general section 51 folder', async () => {
			const response = await request.get(`/applications-service/general-section-51`);

			expect(response.status).toEqual(302);
			expect(response.header.location).toEqual(
				'/applications-service/case/111111111/project-documentation/222222222/s51-advice'
			);
		});
	});
});
