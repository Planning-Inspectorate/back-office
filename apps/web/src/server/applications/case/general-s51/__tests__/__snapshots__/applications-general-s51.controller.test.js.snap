// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General section 51 page #viewGeneralSection51page should render the general section 51 version of the section 51 page 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="top-errors-hook" role="alert"></div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full">
            <h2 class="govuk-heading-l govuk-!-margin-bottom-2">General S51 advice</h2>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible govuk-!-margin-top-2">
            <script nonce="EdcOUaJ8lczj9tIPO0lPow==">
                let clickEventListenerAdded = false
                		window.addEventListener("load", () => {
                			document.querySelector("#pins-subfolder-accordian .govuk-accordion__section-heading").addEventListener("mouseenter", (event) => {
                				if (clickEventListenerAdded) {
                					return
                				}
                				event.target.getElementsByClassName("govuk-accordion__section-button")[0].addEventListener("click", (event) => {
                					if (event.pointerType) {
                						event.currentTarget.blur()
                					}
                				})
                				clickEventListenerAdded = true
                			})
                		})
            </script>
            <div class="govuk-width-container">
                <div class="govuk-grid-row govuk-!-margin-top-0">
                    <div class="govuk-grid-column-two-thirds govuk-!-margin-top-0"><a class="govuk-button govuk-button--secondary govuk-!-margin-0" href="/applications-service/case/111111111/project-documentation/21/s51-advice/create/title"> Create new</a>
                    </div>
                    <div class="govuk-grid-column-one-third"><a class="govuk-link govuk-body-m govuk-!-text-align-right" href="/applications-service/case/111111111/project-documentation/21/s51-advice/publishing-queue"> General S51 publishing queue</a>
                    </div>
                </div>
            </div>
        </div>
    </div>
</main>"
`;
